src/partitioner/local/MetisPartitioner.cpp

```diff
@@ -385,6 +391,139 @@ void MetisPartitioner::createPartitionFiles(std::map<int, int> partMap) {
+            std::vector<size_t> minSumGroup = centralStoreGroups[0];
+            std::vector<int> minPartitionGroup = partitionGroups[0];
+            std::vector<size_t>::iterator minSumGroupIterator;
+            size_t minGroupTotal = 0;
+            int minGroupIndex = 0;
+
+            for (minSumGroupIterator = minSumGroup.begin(); minSumGroupIterator != minSumGroup.end(); ++minSumGroupIterator) {
+                size_t size = *minSumGroupIterator;
+                minGroupTotal += size;
+            }
+
+            for (int loop = 0; loop < 4; loop++) {
+                std::vector<size_t> currentGroup = centralStoreGroups[loop];
+                std::vector<int> currentPartitionGroup = partitionGroups[loop];
+                std::vector<size_t>::iterator currentGroupIterator;
+                size_t currentGroupTotal = 0;
+
+                for (currentGroupIterator = currentGroup.begin(); currentGroupIterator != currentGroup.end(); ++currentGroupIterator) {
+                    int currentSize = *currentGroupIterator;
+                    currentGroupTotal += currentSize;
+                }
+
+                if (currentGroupTotal < minGroupTotal) {
+                    minSumGroup = currentGroup;
+                    minGroupTotal = currentGroupTotal;
+                    minPartitionGroup = currentPartitionGroup;
+                    minGroupIndex = loop;
+                }
+            }
+
+            minSumGroup.push_back(centralStoreSize);
+            centralStoreGroups[minGroupIndex] = minSumGroup;
+
+            minPartitionGroup.push_back(partitionLoop);
+            partitionGroups[minGroupIndex] = minPartitionGroup;
+
+            partitionLoop++;
+        }
+
+        std::map<int, std::vector<int>>::iterator partitionGroupsIterator;
+
+        for (partitionGroupsIterator = partitionGroups.begin() ; partitionGroupsIterator != partitionGroups.end(); ++partitionGroupsIterator) {
+            std::string aggregatePartitionId = "";
+            int group = partitionGroupsIterator->first;
+            std::vector<int> partitionList = partitionGroupsIterator->second;
+
+            if (partitionList.size() > 0) {
+                std::vector<int>::iterator partitionListIterator;
+                std::map<int, std::vector<int>> tempCompositeMap;
+
+                for (partitionListIterator = partitionList.begin(); partitionListIterator != partitionList.end(); ++partitionListIterator) {
+                    int partitionId = *partitionListIterator;
+                    aggregatePartitionId = std::to_string(partitionId) + "_" + aggregatePartitionId;
+
+                    std::map<int, std::vector<int>> currentStorageMap = masterGraphStorageMap[partitionId];
+                    std::map<int, std::vector<int>>::iterator currentStorageMapIterator;
+
+                    for (currentStorageMapIterator = currentStorageMap.begin();currentStorageMapIterator != currentStorageMap.end(); ++currentStorageMapIterator) {
```

Comment:
There are several `for` and `if` statements which are too lengthy. We should break them into two lines.

Reply:
Done